erpnext/hr/doctype/leave_application/leave_application.py

```diff
@@ -40,7 +40,8 @@ def validate(self):
 	def on_update(self):
 		if self.status == "Open" and self.docstatus < 1:
 			# notify leave approver about creation
-			self.notify_leave_approver()
+			if frappe.db.get_single_value("HR Settings", "send_leave_notification"):
```

Comment:
Shouldn't this be also added before calling `notify_employee()`

Reply:
No. Actually, we are not sending any notification if the checkbox is unchecked, to anyone.

Reply:
In `on_submit()` `notify_employee()` is called.
![Screenshot 2021-03-15 at 4 33 46 PM](https://user-images.githubusercontent.com/33727827/111144105-9ecc7280-85ac-11eb-89d0-87377464f507.png)

Again it is called in `on_cancel()`.
![Screenshot 2021-03-15 at 4 34 01 PM](https://user-images.githubusercontent.com/33727827/111144175-bd326e00-85ac-11eb-93a0-643d67557050.png)

And irrespective of if the checkbox is checked or not notification is sent to the employee.
![Screenshot 2021-03-15 at 4 35 02 PM](https://user-images.githubusercontent.com/33727827/111144294-e5ba6800-85ac-11eb-8e26-fb3ca1757c51.png)

Reply:
Got it...